pydantic/_internal/_config.py

```diff
@@ -184,6 +185,15 @@ def __repr__(self):
         c = ', '.join(f'{k}={v!r}' for k, v in self.config_dict.items())
         return f'ConfigWrapper({c})'
 
+    def add_schema_validator_serializer(
+        self, obj: Any, schema: core_schema.CoreSchema, serializer: bool = True
+    ) -> None:
+        core_config = self.core_config(obj)
+        obj.__pydantic_core_schema__ = schema
+        obj.__pydantic_validator__ = create_schema_validator(schema, core_config, self.plugin_settings)
+        if serializer:
+            obj.__pydantic_serializer__ = SchemaSerializer(schema, core_config)
```

Comment:
Why should this go on `ConfigWrapper`? I'm also not sure this abstraction is worth it.

Reply:
It could also just be a free function accepting a config wrapper object (which is used twice here), it doesn't really matter.

Reply:
I think a free-standing function would be better. But since there's already a boolean in here let's not create this abstraction.